src/util/utilization.jl

```diff
@@ -0,0 +1,126 @@
+    d_comp_vec = vcat([[(comp_type,comp_id) for comp_id in comp_ids] for (comp_type,comp_ids) in get_repairable_components(data)]...)
+    d_comp_cost = [_util_value(data,comp_type,comp_id) for (comp_type,comp_id) in d_comp_vec]
+    d_comp_vec = [d_comp_vec[i] for i in sortperm(d_comp_cost)] # reordered damaged component vector
+
+
+    restoration_period = Dict{Tuple{String, String},Any}(
+        (d_comp_vec[id])=>id for id in 1:length(d_comp_vec)
+    )
+
+    # apply precedent repair requirements
+    repair_constraints = _calculate_repair_precedance(data)
+    updated = true
+    while updated
+        updated = false
+        for (r_comp, precedance_comps) in repair_constraints
+            if r_comp in d_comp_vec
+                precendent_repair_periods = [get(restoration_period,pr_comp,0) for pr_comp in precedance_comps]
+                if !isempty(precendent_repair_periods)
+                    final_precedent_repair = maximum(precendent_repair_periods)
+                else
+                    final_precedent_repair = 0
+                end
+                if restoration_period[r_comp] < final_precedent_repair
+                    Memento.info(_PM._LOGGER, "Changing $r_comp repair from $(restoration_period[r_comp]) to $final_precedent_repair")
+                    updated = true
+                    restoration_period[r_comp] = final_precedent_repair
+                end
+            end
+        end
+    end
+
+    restoration_order = Dict{String,Any}("$nwid"=>Tuple{String, String}[] for nwid in 1:length(d_comp_vec))
+    for ((comp_type,comp_id),nwid) in restoration_period
+        push!(restoration_order["$(nwid)"],(comp_type, comp_id))
+    end
+
+    return restoration_order
+end
+
+function _util_value(data::Dict{String,<:Any}, comp_type::String, comp_id::String)
+    if comp_type=="branch"
+        return _util_value_branch(data,comp_id)
+    elseif comp_type=="gen"
+        return _util_value_gen(data,comp_id)
+    elseif comp_type=="bus"
+        return _util_value_bus(data,comp_id)
+    elseif comp_type=="storage"
+        return _util_value_storage(data,comp_id)
+    else
+        Memento.error(_PM._LOGGER, "Component $comp_type does not have a supported utilization cost. Setting cost to NaN")
+        return NaN
+    end
+end
+
+function _util_value_branch(data::Dict{String,<:Any}, comp_id::String)
+    return data["branch"][comp_id]["rate_a"]
+end
+
+function _util_value_gen(data::Dict{String,<:Any}, comp_id::String)
+    return data["gen"][comp_id]["pmax"]
+end
+
+function _util_value_bus(data::Dict{String,<:Any}, comp_id::String)
+    bus_load = 0.0 # this is orders of mag slower than the other util_value functions
+    for (load_id,load) in data["load"]
+        if load["load_bus"]==parse(Int,comp_id::String)
+            bus_load+=load["pd"]
+        end
+    end
+    return bus_load
+end
+
+function _util_value_storage(data::Dict{String,<:Any}, comp_id::String)
+    return data["storage"][comp_id]["energy"]
+end
```

Comment:
Are these functions  `_util_value_xyz` used in other places (or planned to be reused some other places)?  If not, I suggest to move the code from these functions into `_util_value`, I think it will make that function easer to follow and understand.